TaskManager/TaskManager/WeeklyTaskService.cs

```diff
@@ -0,0 +1,162 @@
+            for (int i = 0; i < _counter; i++)
+            {
+                PrintTask(_tasks[i], i);
+            }
+        }
+
+        public void HandleEdit()
+        {
+            Console.WriteLine("Input number to edit:");
+            var inputNumber = Console.ReadLine();
+            var taskNumber = int.Parse(inputNumber);
+
+            Console.WriteLine("Input new task data in format: {},{},{}:");
+            var inputTaskData = Console.ReadLine();
+            var task = ParseNewTask(inputTaskData);
+            _tasks[taskNumber - 1] = task;
+        }
+
+        public void HandleFilterByDate()
+        {
+            Console.WriteLine("Input date:");
+            var inputData = Console.ReadLine();
+            var date = DateTime.Parse(inputData);
+            
+            for (int i = 0; i < _counter; i++)
+            {
+                var task = _tasks[i];
+                if (task.GetDate() >= date)
+                {
+                    PrintTask(task, i);
+                }
+            }
+        }
+
+        private static void PrintTask(WeeklyTask task, int i)
+        {
+            Console.WriteLine(task.ConvertToString(i));
+        }
+
+        public void HandleFilterByPriority()
+        {
+            Console.WriteLine("Input the priority number:");
+            var inputData = Console.ReadLine();
+            var priority = Enum.Parse<Priority>(inputData);
+            if (priority is >= 0 or < (Priority)4)
+            {
+                for (int counterOfPrintTask = 0, i = 0; i < _counter; i++)
+                {
+                    var task = _tasks[i];
+                    if (task.GetPriority() == priority)
+                    {
+                        PrintTask(task, i);
+                        counterOfPrintTask++;
+                    }
+                    if ((_counter - 1 == i) && (counterOfPrintTask == 0))
+                    {
+                        Console.WriteLine($"Tasks with priority {priority} not set");
+                    }
+                }
+            }
+            else
+            {
+                Console.WriteLine("Invalid priority number! Remember, you must only enter numbers from 0 to 3! \nTry again:");
+            }
+        }
+
+        private WeeklyTask ParseNewTask(string? inputData)
+        {
+            var parts = inputData?.Split(",");
+
+            if (parts == null || parts.Length < 1 || parts.Length > 4)
+            {
+                Console.WriteLine("Invalid task format, try again");
+                return null;
+            }
+
+            return CreateNewTask(parts);
+        }
+
+        private WeeklyTask CreateNewTask(string[] parts)
+        {
+            return parts.Length switch
+            {
+                1 => CreateTaskWithName(parts),
+                2 => CreateTaskWithDate(parts),
+                3 => CreateTaskWithDateTime(parts),
+                4 => CreateTaskWithDateAndPriority(parts),
+                _ => null,
```

Comment:
when you return null, ensure all code above checks this null value - otherwise it will throw NRE